test/test_utils.py

```diff
@@ -711,6 +715,72 @@ def test_unmarshal_tools_pcr_values(self):
         self.assertEqual(digs[1], b"\x00" * 20)
         self.assertEqual(digs[2], b"\xFF" * 32)
 
+    def test_credential_to_tools(self):
+        id_object_len = 40
+        encrypted_secret_len = 68
+
+        id_object_bytes = os.urandom(id_object_len)
+        encrypted_secret_bytes = os.urandom(encrypted_secret_len)
+
+        id_object = TPM2B_ID_OBJECT(credential=id_object_bytes)
+        encrypted_secret = TPM2B_ENCRYPTED_SECRET(secret=encrypted_secret_bytes)
+
+        credential_blob = credential_to_tools(id_object, encrypted_secret)
+
+        self.assertEqual(credential_blob[0:4], int(0xBADCC0DE).to_bytes(4, "big"))
+        self.assertEqual(credential_blob[4:8], int(1).to_bytes(4, "big"))
+        self.assertEqual(credential_blob[8:10], int(id_object_len).to_bytes(2, "big"))
+        self.assertEqual(credential_blob[10 : 10 + id_object_len], id_object_bytes)
+
+        self.assertEqual(
+            int.from_bytes(
+                credential_blob[10 + id_object_len : 12 + id_object_len],
+                byteorder="big",
+            ),
+            encrypted_secret_len,
+        )
+        self.assertEqual(
+            credential_blob[
+                12 + id_object_len : 12 + id_object_len + encrypted_secret_len
+            ],
+            encrypted_secret_bytes,
+        )
+
+    def test_tools_to_credential(self):
+        id_object_len = 40
+        encrypted_secret_len = 68
+
+        id_object_bytes = os.urandom(id_object_len)
+        encrypted_secret_bytes = os.urandom(encrypted_secret_len)
+
+        data = b""
```

Comment:
bytearray and use extend to add to it, as bytes is immutable so were constantly creating a destroying objects here